android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -243,56 +256,67 @@ public MatchingEngineHelper(final Builder builder) {
      */
     protected void initEdgeEventsConfigPrefs() {
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mActivity);
-        onSharedPreferenceChanged(prefs, prefKeyLocationUpdatePattern);
-        onSharedPreferenceChanged(prefs, prefKeyLocationUpdateInterval);
-        onSharedPreferenceChanged(prefs, prefKeyLocationMaxNumUpdates);
-        onSharedPreferenceChanged(prefs, prefKeyFindCloudletEventTrigger);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyTestType);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyUpdatePattern);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyTestTriggerMode);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyUpdateInterval);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyMaxNumUpdates);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyTestPort);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyThreshold);
-        onSharedPreferenceChanged(prefs, prefKeyPerfMarginSwitch);
-        onSharedPreferenceChanged(prefs, prefKeyAutoMigration);
+        onEdgeEventPreferenceChanged(prefs, prefKeyEnableEdgeEvents);
+        onEdgeEventPreferenceChanged(prefs, prefKeyOverrideDefaultConfig);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLocationUpdatePattern);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLocationUpdateInterval);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLocationMaxNumUpdates);
+        onEdgeEventPreferenceChanged(prefs, prefKeyFindCloudletEventTrigger);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestType);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyUpdatePattern);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestTriggerMode);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyUpdateInterval);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyMaxNumUpdates);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestPort);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyThreshold);
+        onEdgeEventPreferenceChanged(prefs, prefKeyPerfMarginSwitch);
+        onEdgeEventPreferenceChanged(prefs, prefKeyAutoMigration);
     }
 
-    public void startEdgeEvents() {

```

Comment:
@lgarner, can give this method a little extra scrutiny please? Maybe view as raw to see the flow more clearly without the diffs in view.